ide/coqide/coqide.ml

```diff
@@ -89,36 +89,21 @@ let make_coqtop_args fname =
     | Ignore_args -> !sup_args
     | Append_args -> !sup_args
     | Subst_args -> [] in
-  let proj, args =
-    if read_project#get = Ignore_args then "", base_args
-    else
-      match !custom_project_file, fname with
-      | Some (d,proj), _ -> d, coqtop_args_from_project proj @ base_args
-      | None, None -> "", base_args
-      | None, Some the_file ->
-        match
-          CoqProject_file.find_project_file
-            ~from:(Filename.dirname the_file)
-            ~projfile_name:project_file_name#get
-        with
-        | None -> "", base_args
-        | Some proj ->
-          let warning_fn x = Feedback.msg_warning Pp.(str x) in
-          proj, coqtop_args_from_project (read_project_file ~warning_fn proj) @ base_args
-  in
-  let args = match fname with
-    | None -> args
-    | Some fname ->
-      if List.exists (String.equal "-top") args then args
-      else
-        (* We basically copy the code of Names.check_valid since it is not exported *)
-        (* to coqide. This is to prevent a possible failure of parsing  "-topfile"  *)
-        (* at initialization of coqtop (see #10286) *)
-        match Unicode.ident_refutation (Filename.chop_extension (Filename.basename fname)) with
-        | Some (_,x) -> output_string stderr (x^"\n"); exit 1
-        | None -> "-topfile"::fname::args
```

Comment:
Doesn't this mean it stops passing -topfile? instead you could just ignore the error ie
~~~
        | Some (_,x) -> args
        | None -> "-topfile"::fname::args
~~~

Reply:
You’re right. Thanks.